### PR TITLE
All route ResolvedRefs reasons added to docs

### LINKED ISSUE #1313: Missing reasons in the `ResolvedRefs` `RouteConditionType` documentation
**What happened**:
The `ResolvedRefs` Route condition (https://github.com/kubernetes-sigs/gateway-api/blob/main/apis/v1beta1/shared_types.go#L227-L234) documentation does not include the reasons `InvalidKind` and `BackendNotFound`  among the possible ones that make it False.

What you expected to happen:

Since this godoc section's purpose is to enumerate the possible reasons that may lead to having one `ConditionStatus` instead of another, it is needed to ensure its exhaustivity. Hence, `InvalidKind` and `BackendNotFound` should be added to the possible reasons that can make the Accepted route condition False in the godoc section.

How to reproduce it (as minimally and precisely as possible):

Anything else we need to know?:

### PR DESCRIPTION
**Pull request by mlavacca**
**What type of PR is this?**

/kind documentation

**What this PR does / why we need it**:
`InvalidKind` and `BackendNotFound` are added to the possible failure reasons of the `ResolvedRefsCondition` in the route godoc.

**Which issue(s) this PR fixes**:

Fixes #1313 

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: apis/v1alpha2/shared_types.go [hunk 1]
```diff
@@ -331,10 +333,9 @@ type RouteStatus struct {
 // Hostname is the fully qualified domain name of a network host. This matches
 // the RFC 1123 definition of a hostname with 2 notable exceptions:
 //
-// 1. IPs are not allowed.
-// 2. A hostname may be prefixed with a wildcard label (`*.`). The wildcard
-//
-//	label must appear by itself as the first label.
+//  1. IPs are not allowed.
+//  2. A hostname may be prefixed with a wildcard label (`*.`). The wildcard
+//     label must appear by itself as the first label.
```
Comment (robscott) [inline apis/v1alpha2/shared_types.go:338]: Why the extra spaces here? If I'm understanding this correctly, we're moving from one space after `//` to two, is that intentional?

**Review by robscott — COMMENTED** [review]
Thanks @mlavacca!

File: apis/v1alpha2/shared_types.go (hunk 1 above)
Comment (mlavacca) [reply apis/v1alpha2/shared_types.go]: It was unintentional, `go fmt` somehow produced that. I noticed that the "verify" job failed. I have updated the `yaml` files, should be good now.

**Comment by robscott** [comment]
Thanks @mlavacca!

/lgtm
/approve

**Pull request merged**